Description: Build against system libs instead of vendored libraries
 This patch changes the QuickBuild toolchain to link to the system versions of:
   - libdiscord-rpc
   - libFLAC
   - liblua5.3
   - libzstd
   - SPIRV-Cross
 …rather than the vendored versions of those libraries present in the tarball
 and typically compiled at build time. The resulting retroarch binary produced
 is roughly 1 MiB smaller, on average.
Author: Peter J. Mello <admin@petermello.net>
Origin: other, https://git.launchpad.net/~libretro/libretro/+git/RetroArch-debian/commit/patches/0000-use-system-libs.patch?id=00fcf2a6c54be09d8b84443651063c7527f420db
Bug-Debian: https://bugs.debian.org/949708
Bug-Debian: https://bugs.debian.org/1053190
Forwarded: no
Last-Update: 2025-07-19

--- a/Makefile.common
+++ b/Makefile.common
@@ -1895,14 +1895,9 @@ endif
 
 ifeq ($(HAVE_SPIRV_CROSS), 1)
    DEFINES += -DHAVE_SPIRV_CROSS
-   INCLUDE_DIRS += -I$(DEPS_DIR)/SPIRV-Cross
-   OBJ += $(DEPS_DIR)/SPIRV-Cross/spirv_cross.o
-   OBJ += $(DEPS_DIR)/SPIRV-Cross/spirv_cfg.o
-   OBJ += $(DEPS_DIR)/SPIRV-Cross/spirv_glsl.o
-   OBJ += $(DEPS_DIR)/SPIRV-Cross/spirv_hlsl.o
-   OBJ += $(DEPS_DIR)/SPIRV-Cross/spirv_msl.o
-   OBJ += $(DEPS_DIR)/SPIRV-Cross/spirv_parser.o
-   OBJ += $(DEPS_DIR)/SPIRV-Cross/spirv_cross_parsed_ir.o
+   INCLUDE_DIRS += -I/usr/include/spirv_cross
+   LIBS += -lspirv-cross-cpp -lspirv-cross-msl -lspirv-cross-glsl \
+			-lspirv-cross-c-shared -lspirv-cross-core
 endif
 
 ifeq ($(WANT_WGL), 1)
@@ -1946,31 +1941,9 @@ ifeq ($(HAVE_7ZIP),1)
 endif
 
 ifeq ($(HAVE_ZSTD),1)
-   INCLUDE_DIRS += -I$(DEPS_DIR)/zstd/lib
+   INCLUDE_DIRS += -I/usr/include
    DEFINES += -DHAVE_ZSTD -DZSTD_DISABLE_ASM
-   ZSOBJ = $(DEPS_DIR)/zstd/lib/common/entropy_common.o \
-           $(DEPS_DIR)/zstd/lib/common/error_private.o \
-           $(DEPS_DIR)/zstd/lib/common/fse_decompress.o \
-           $(DEPS_DIR)/zstd/lib/common/zstd_common.o \
-           $(DEPS_DIR)/zstd/lib/common/xxhash.o \
-           $(DEPS_DIR)/zstd/lib/compress/fse_compress.o \
-           $(DEPS_DIR)/zstd/lib/compress/hist.o \
-           $(DEPS_DIR)/zstd/lib/compress/huf_compress.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_compress.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_compress_literals.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_compress_sequences.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_compress_superblock.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_double_fast.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_fast.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_lazy.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_ldm.o \
-           $(DEPS_DIR)/zstd/lib/compress/zstd_opt.o \
-           $(DEPS_DIR)/zstd/lib/decompress/huf_decompress.o \
-           $(DEPS_DIR)/zstd/lib/decompress/zstd_ddict.o \
-           $(DEPS_DIR)/zstd/lib/decompress/zstd_decompress.o \
-           $(DEPS_DIR)/zstd/lib/decompress/zstd_decompress_block.o
-
-   OBJ +=  $(ZSOBJ)
+   LIBS += -lzstd
 endif
 
 ifeq ($(HAVE_IBXM), 1)
@@ -2007,6 +1980,7 @@ ifeq ($(HAVE_BUILTINFLAC),1)
    OBJ += $(FLACOBJ)
 else ifeq ($(HAVE_FLAC),1)
    DEFINES += -DHAVE_FLAC
+   INCLUDE_DIRS += -I/usr/include
    LIBS    += $(FLAC_LIBS)
 endif
 
@@ -2392,41 +2366,10 @@ ifeq ($(HAVE_NETWORKING), 1)
 
       ifeq ($(HAVE_LUA), 1)
          DEFINES += -DHAVE_LUA \
-                    -DLUA_32BITS
-         INCLUDE_DIRS += -Ideps/lua/src
-         OBJ += deps/lua/src/lapi.o \
-                deps/lua/src/lcode.o \
-                deps/lua/src/lctype.o \
-                deps/lua/src/ldebug.o \
-                deps/lua/src/ldo.o \
-                deps/lua/src/ldump.o \
-                deps/lua/src/lfunc.o \
-                deps/lua/src/lgc.o \
-                deps/lua/src/llex.o \
-                deps/lua/src/lmem.o \
-                deps/lua/src/lobject.o \
-                deps/lua/src/lopcodes.o \
-                deps/lua/src/lparser.o \
-                deps/lua/src/lstate.o \
-                deps/lua/src/lstring.o \
-                deps/lua/src/ltable.o \
-                deps/lua/src/ltm.o \
-                deps/lua/src/lundump.o \
-                deps/lua/src/lvm.o \
-                deps/lua/src/lzio.o \
-                deps/lua/src/lauxlib.o \
-                deps/lua/src/lbaselib.o \
-                deps/lua/src/lbitlib.o \
-                deps/lua/src/lcorolib.o \
-                deps/lua/src/ldblib.o \
-                deps/lua/src/liolib.o \
-                deps/lua/src/lmathlib.o \
-                deps/lua/src/loslib.o \
-                deps/lua/src/lstrlib.o \
-                deps/lua/src/ltablib.o \
-                deps/lua/src/lutf8lib.o \
-                deps/lua/src/loadlib.o \
-                deps/lua/src/linit.o
+                    -DLUA_32BITS \
+                    -DLUA_USE_POSIX
+         INCLUDE_DIRS += -I/usr/include/lua5.3
+         LIBS += -llua5.3 -lm -ldl
       else
          DEFINES += -DRC_DISABLE_LUA
       endif
@@ -2437,15 +2380,13 @@ ifeq ($(HAVE_NETWORKING), 1)
       NEED_CXX_LINKER = 1
       HAVE_PRESENCE = 1
       DEFINES += -DHAVE_DISCORD
-      INCLUDE_DIRS += -Ideps/discord-rpc/include
+      INCLUDE_DIRS += -I/usr/include/DiscordRPC
 
       ifneq ($(HAVE_THREADS), 1)
          DEFINES += -DDISCORD_DISABLE_IO_THREAD
       endif
 
-      OBJ     += deps/discord-rpc/src/discord_rpc.o \
-                 deps/discord-rpc/src/rpc_connection.o \
-                 deps/discord-rpc/src/serialization.o
+      LIBS += -ldiscord-rpc
 
       OBJ     += network/discord.o
 
@@ -2454,10 +2395,6 @@ ifeq ($(HAVE_NETWORKING), 1)
                  deps/discord-rpc/src/connection_win.o
          LIBS += -lpsapi -ladvapi32
       endif
-      ifneq ($(findstring Linux,$(OS)),)
-         OBJ  += deps/discord-rpc/src/discord_register_linux.o \
-                 deps/discord-rpc/src/connection_unix.o
-      endif
       ifneq ($(findstring Darwin,$(OS)),)
          OBJ  += deps/discord-rpc/src/discord_register_osx.o \
                  deps/discord-rpc/src/connection_unix.o
--- a/libretro-common/formats/libchdr/libchdr_flac.c
+++ b/libretro-common/formats/libchdr/libchdr_flac.c
@@ -17,6 +17,8 @@
 #define DR_FLAC_IMPLEMENTATION
 #define DRFLAC_API static INLINE
 #include <dr/dr_flac.h>
+#else
+#include <FLAC/stream_decoder.h>
 #endif
 
 /***************************************************************************
@@ -25,8 +27,12 @@
  */
 
 static size_t flac_decoder_read_callback(void *userdata, void *buffer, size_t bytes);
+#ifdef HAVE_DR_FLAC
 static drflac_bool32 flac_decoder_seek_callback(void *userdata, int offset, drflac_seek_origin origin);
 static void flac_decoder_metadata_callback(void *userdata, drflac_metadata *metadata);
+#else
+static void flac_decoder_metadata_callback(const FLAC__StreamDecoder *decoder, const FLAC__StreamMetadata *metadata, void *client_data);
+#endif
 static void flac_decoder_write_callback(void *userdata, void *buffer, size_t bytes);
 
 
@@ -64,10 +70,15 @@ int flac_decoder_init(flac_decoder *deco
 
 void flac_decoder_free(flac_decoder* decoder)
 {
+#ifdef HAVE_DR_FLAC
 	if ((decoder != NULL) && (decoder->decoder != NULL)) {
 		drflac_close(decoder->decoder);
 		decoder->decoder = NULL;
 	}
+#else
+	if ((decoder != NULL) && (decoder->decoder != NULL))
+		FLAC__stream_decoder_delete(decoder->decoder);
+#endif
 }
 
 /*-------------------------------------------------
--- a/network/discord.h
+++ b/network/discord.h
@@ -24,7 +24,6 @@
 #include <boolean.h>
 
 #include <discord_rpc.h>
-#include "../deps/discord-rpc/include/discord_rpc.h"
 #include "presence.h"
 
 /* The Discord API specifies these variables:
